### PR TITLE
DOCS-6705 OTel/APM guide for custom instrumentation interop and trace context/IDs

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do? What is the motivation?
DOCS-6705 and DOCS-6703 - new guide explaining how and why of custom instrumenting code with OTel API, how context and trace IDs are propagated and interoperable. Links.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

**Review by pierotibou — APPROVED** [review]
Good for .NET

File: content/en/tracing/trace_collection/otel_instrumentation/go.md [hunk 1]
```diff
@@ -23,10 +26,10 @@ The following OpenTelemetry features are implemented in the Datadog library as n
 
 | Feature                               | Support notes                       |
 |---------------------------------------|------------------------------------|
-| [OpenTelemetry Context propagation][1]         | [Datadog distributed header format][9] is used instead. |
+| [OpenTelemetry Context propagation][1]         | [W3C and Datadog header formats][9] are used. |
```
[the remark this answers is not in the document]
Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/otel_instrumentation/go.md:29]: Oh another comment, maybe we can say that these are enabled by default. So taking the previous comment and this comment together, it would look like the following. What do you think?
```suggestion
| [OpenTelemetry Context propagation][1]         | [W3C Trace Context and Datadog header formats][9] are enabled, by default. |
```

File: content/en/tracing/trace_collection/otel_instrumentation/go.md [hunk 2]
```diff
@@ -23,10 +26,10 @@ The following OpenTelemetry features are implemented in the Datadog library as n
...
 | [Span processors][2]                  | Unsupported                                          |
 | [Span Exporters][3]                   | Unsupported                                            |
-| Trace/span [ID generators][4]         | ID generation is performed by `ddtrace`.           |
+| Trace/span [ID generators][4]         | [128-bit ID generation][12] is performed by `ddtrace`.           |
```
Comment (zacharycmontoya) [inline content/en/tracing/trace_collection/otel_instrumentation/go.md:32]: Trace ID's and Span ID's have different sizes, so maybe we can reword it to something like this? I also substituted `ddtrace` with `tracing library` so we could copy/paste this description to other languages, if you think that's okay.

```suggestion
| Trace/span [ID generators][4]         | ID generation is performed by the tracing library, with support for [128-bit trace ID's][12].           |
```

File: content/en/tracing/trace_collection/trace_context_propagation/_index.md [hunk 3]
```diff
@@ -11,8 +11,25 @@ further_reading:
...
+      text: 'Interoperability of OpenTelemetry API and Datadog instrumented traces'
 ---
 
+W3C trace propagation and 128 bit headers is available for all supported languages. The default trace propagation style for each language is:
+
+| Language | Default inject and extract setting |
+| -------- | ---------------------------------- |
+| Java     | `datadog,tracecontext`             |
+| Python   | `tracecontext,datadog`             |
+| Ruby     | `datadog,tracecontext`             |
```
Comment (zacharycmontoya) [inline content/en/tracing/trace_collection/trace_context_propagation/_index.md:25]: It looks like Ruby by default will _extract_ `Datadog,b3multi,b3,tracecontext` and then _inject_ `Datadog,tracecontext`. Since the extract operation is the one that affects propagation the most, I suggest this small update:

```suggestion
| Ruby     | `Datadog,b3multi,b3,tracecontext`             |
```

File: content/en/tracing/trace_collection/trace_context_propagation/php.md [hunk 4]
```diff
@@ -26,9 +29,9 @@ You can use the following environment variables to configure the PHP tracing lib
 - `DD_TRACE_PROPAGATION_STYLE_INJECT=Datadog,tracecontext,B3 single header`
 - `DD_TRACE_PROPAGATION_STYLE_EXTRACT=Datadog,tracecontext,B3 single header`
 
-The environment variable values are comma-separated lists of header styles enabled for injection or extraction. By default, only the `tracecontext` and `Datadog` injection styles are enabled.
+The environment variable values are comma-separated lists of header styles enabled for injection or extraction. The default style setting is `tracecontext,Datadog`.
 
-If multiple extraction styles are enabled, the extraction attempt is completed with the following priorities: `tracecontext` has priority, then `Datadog`, then B3.
+If multiple extraction styles are enabled, the extraction attempt is completed with the following priorities: `tracecontext` has priority, then `Datadog`, then `B3`.
```
Comment (zacharycmontoya) [inline content/en/tracing/trace_collection/trace_context_propagation/php.md:34]: @bwoebi can you confirm whether the extraction is now order-dependent? If that's the case, then we should be able to remove this line from the documentation, right?

Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/trace_context_propagation/php.md:34]: Or we could take a line from the Ruby documentation that says: "If multiple extraction styles are enabled extraction attempt is done on the order those styles are configured and first successful extracted value is used."

Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/trace_context_propagation/php.md:34]: Or we could change it to the following line, taken from the Java page:

When multiple extraction styles are enabled, the extraction attempt is done on the order those styles are configured, using the first successful extracted value. If later valid trace contexts are found, they are terminated and appended as span links. Moreover, if the `tracecontext` style is enabled, W3C Tracestate is propagated if W3C Traceparent matches the extracted context.

File: content/en/tracing/trace_collection/trace_context_propagation/_index.md (hunk 3 above)
Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/trace_context_propagation/_index.md:25]: @marcotc can you also confirm this change?

File: content/en/tracing/trace_collection/trace_context_propagation/_index.md [hunk 5]
```diff
@@ -11,8 +11,25 @@ further_reading:
...
+| Go       | `tracecontext,datadog`             |
+| Node.js  | `datadog,tracecontext`             |
+| PHP      | `tracecontext,datadog`             |
+| .NET     | `tracecontext,datadog`             |
+| C++      | `datadog,tracecontext` or with Envoy or nginx proxies: `tracecontext,Datadog` |
```
Comment (zacharycmontoya) [inline content/en/tracing/trace_collection/trace_context_propagation/_index.md:30]: Would it be okay for the Envoy and nginx proxies to have their own line in this table? cc @dgoffredo

File: content/en/tracing/trace_collection/trace_context_propagation/cpp.md
Comment (zacharycmontoya) [inline content/en/tracing/trace_collection/trace_context_propagation/cpp.md:1]: @dgoffredo can you review the changes on this page? I don't know where the cpp documentation stands since it's complicated

File: content/en/opentelemetry/guide/otel_api_tracing_interoperability.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+- The traces produced by your running code can be processed, analyzed, and monitored alongside Datadog traces and in Datadog proprietary products such as [Continuous Profiler][3], [Data Streams Monitoring][4], [Application Security Management][5], and [Live Processes][6].
+
+## W3C trace context propagation
+
+To facilitate seamless handling of OpenTelemetry trace data within Datadog, and correlating it with trace data generated by Datadog instrumentation, the most recent versions of Datadog tracing libraries support both [Datadog (`datadog`) and W3C (`tracecontext`) propagation styles][8] by default. [Update your runtime tracing library dependencies][7] to the latest version.
+
+Using this context propagation style allows Datadog tracers to operate in the same application environment with OpenTelemetry SDKs and other W3C compliant tracers.
+
+
+## 128-bit trace IDs
```
Comment (zacharycmontoya) [inline content/en/opentelemetry/guide/otel_api_tracing_interoperability.md:32]: In this section can we specifically say "trace IDs" everywhere we refer to an "ID"? In OpenTelemetry, trace ID's are 128-bit and span ID's are 64-bit, so I want to avoid any confusion

File: content/en/opentelemetry/guide/otel_api_tracing_interoperability.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  text: "Custom Instrumentation with OpenTelemetry API"
+- link: "/tracing/trace_collection/trace_context_propagation/"
+  tag: "Documentation"
+  text: "Trace Context Propagation in Datadog"
+
+---
+
+## Overview
+
+Datadog tracing libraries provide an implementation of the [OpenTelemetry API][1] for instrumenting your code. This means you can maintain vendor-neutral instrumentation of all your services, while still taking advantage of Datadog's native implementation, features, and products.
```
Comment (zacharycmontoya) [inline content/en/opentelemetry/guide/otel_api_tracing_interoperability.md:16]: One small detail that I'd like for the documentation to call out is that the user would replace the OpenTelemetry SDK with the Datadog tracing library in the instrumented application. I'm not sure what the wording would look like though

File: content/en/tracing/trace_collection/trace_context_propagation/_index.md (hunk 3 above)
Comment (marcotc) [reply content/en/tracing/trace_collection/trace_context_propagation/_index.md:25]: This is correct for Ruby, thank you @zacharycmontoya!

File: content/en/tracing/trace_collection/trace_context_propagation/ruby.md [hunk 8]
```diff
@@ -36,7 +39,7 @@ The value of the environment variable is a comma-separated list of header styles
 
 If multiple extraction styles are enabled extraction attempt is done on the order those styles are configured and first successful extracted value is used.
 
-The default extraction styles are, in order, `Datadog`, `b3multi`, and `b3`.
+The default extraction styles are, in order, `Datadog`, `tracecontext`, `b3multi`, and `b3`.
```
[the remark this answers is not in the document]
Comment (marcotc) [reply content/en/tracing/trace_collection/trace_context_propagation/ruby.md:42]: This is correct for Ruby, thank you @zacharycmontoya!

File: content/en/tracing/trace_collection/trace_context_propagation/php.md (hunk 4 above)
Comment (zacharycmontoya) [reply content/en/tracing/trace_collection/trace_context_propagation/php.md:34]: I got feedback from Bob offline, it's now done in the specified order, so we can change the text to the following:

```suggestion
If multiple extraction styles are enabled, the extraction attempt is done on the order those styles are configured and first successful extracted value is used.
```

File: content/en/opentelemetry/guide/otel_api_tracing_interoperability.md (hunk 7 above)
Comment (kayayarai) [reply content/en/opentelemetry/guide/otel_api_tracing_interoperability.md:16]: Good idea. With my next commit, I took the last bulled from the list below, and changed it a bit to include this piece of info.

Comment (zacharycmontoya) [reply content/en/opentelemetry/guide/otel_api_tracing_interoperability.md:16]: Looks good!

File: content/en/opentelemetry/guide/otel_api_tracing_interoperability.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+- link: "/tracing/trace_collection/otel_instrumentation/"
+  tag: "Documentation"
+  text: "Custom Instrumentation with OpenTelemetry API"
+- link: "/tracing/trace_collection/trace_context_propagation/"
+  tag: "Documentation"
+  text: "Trace Context Propagation in Datadog"
+
+---
+
+## Overview
```
Comment (zacharycmontoya) [inline content/en/opentelemetry/guide/otel_api_tracing_interoperability.md:14]: One last little bit of non-blocking feedback: If we end up making this page more of a landing page to indicate how Datadog tracing libraries interoperate with OpenTelemetry components, then we could rename this overview section to a section like `Custom instrumentation with the OpenTelemetry API`

**Review by zacharycmontoya — APPROVED** [review]
Thanks Kari! The updates look good across languages and everything appears up-to-date with our latest releases now

File: layouts/shortcodes/otel-custom-instrumentation.md [hunk 10]
```diff
@@ -1,6 +1,16 @@
...
-If your code is custom instrumented with the [OpenTelemetry (OTel) API][101], you can configure it to generate Datadog-style spans and traces to be processed by the Datadog tracing library for your language, and send those to Datadog. 
+Datadog tracing libraries provide an implementation of the [OpenTelemetry API][101] for instrumenting your code. This means you can maintain vendor-neutral instrumentation of all your services, while still taking advantage of Datadog's native implementation, features, and products. You can configure it to generate Datadog-style spans and traces to be processed by the Datadog tracing library for your language, and send those to Datadog.
+
+By instrumenting your code with OpenTelemetry API:
+
+- Your code remains free of vendor-specific API calls.
+- Your code does not depend on Datadog tracing libraries at compile time (only runtime).
+- Your code does not use the deprecated OpenTracing API.
+
+Replace the OpenTelemetry SDK with the Datadog tracing library in the instrumented application, and the traces produced by your running code can be processed, analyzed, and monitored alongside Datadog traces and in Datadog proprietary products
```
Comment (apigirl) [inline layouts/shortcodes/otel-custom-instrumentation.md:11]:
```suggestion
Replace the OpenTelemetry SDK with the Datadog tracing library in the instrumented application, and the traces produced by your running code can be processed, analyzed, and monitored alongside Datadog traces and in Datadog proprietary products.
```

**Review by apigirl — APPROVED** [review]
great PR! Just one tiny nit is that the order here 
image
is different than the order here 
image
This could be an iterative update though
